docs/Custom_Device_Types.md

```diff
@@ -0,0 +1,184 @@
+
+1.	Initialize
+2.	Start
+3.	Read Data from Hardware
+4.	Write Data to Hardware
+5.	Close
+
+This custom device runs in-line with the PCL, which calls each case at a specific time with respect to the other components in the VeriStand engine. The PCL will not proceed until the custom device case has completed.
+
+### Initialize
+
+The Initialize case executes before the PCL starts. In this case, you can read the device configuration information from properties using the reference to the device. Initialize data and buffers used internally in the device. The framework compiles the list of Data References for the custom device Inputs and Outputs in advance using **[Custom Device API](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_custom_device_api_vis_pal/)** » **[Driver Functions](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_driver_functions_vis_pal/)** » **[Get Custom Device Channel List](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_get_custom_device_channel_list_vi/)** and Custom Device API.lvlib » Templates » RT Driver VIs » Inline » Inline Driver Utilities » Channel Data References » Get Channel Data Reference.
+ 
+![](images/Initialize_State_of_the_Inline_Hardware.jpg)
+
+**Figure: Initialize State of the Inline Hardware Interface Framework**
+
+Since the PCL hasn’t started yet, you can't read or write channel values in the Initialize case.
+
+### Start
+
+The Start case executes after Initialization and before the PCL starts running. There’s no difference between what code you can place in the Initialize and Start states. Since the PCL hasn’t started yet, you can't read or write channel values in the Start case.
+
+### Read Data from HW
+
+The Read Data from HW case executes at the beginning of the PCL, before other components (such as stimulus profiles, faults, alarms, procedures, etc.) execute. For a detailed timing diagram, see the **[Outline of PCL Iteration](https://niveristand-custom-device-handbook.readthedocs.io/en/latest/Custom_Device_Types.html#outline-of-pcl-iteration)** section. After processing system mappings, the data obtained in this case is available to the other components of the system for the remainder of the PCL iteration.
```

Comment:
I'm reviewing from this link: https://niveristand-custom-device-handbook--10.org.readthedocs.build/en/10/Custom_Device_Types.html

and the link to "Outline of PCL Iteration" goes to a broken page for me.
Not sure if that's an issue with the 'built page' or is just something related to this being a PR under review.

Reply:
It will work when the "Outline of PCL Iteration" will be merged into main.